### DISCUSSION TITLE
Implementing tests

### DISCUSSION THREAD

**Discussion by ali-thowfeek** [ali-thowfeek/simple_html_css_flutter#11] [Ideas]
@nohli I think it's time we write some tests on to the package. If I could get a little push with this, I could gradually work on it.

**Comment by nohli** [comment]
@ali-thowfeek sounds good! What do you mean by push?

Also, I would like to add lint to the package and make the code cleaner.

FYI Right now, RichText(text: HTML.toTextSpan(context, string) with a html list (ul) in the string returns \n\n at the end. Probably it would be better without trailing blank space.
Also, an arrow (& rarr; without the space) isn't converted, although it appears in the package html_unescape (https://github.com/filiph/html_unescape/blob/bec0a1d683cda1ec06fc1c923abe9419ef707611/lib/src/data/named_chars_all.dart)

**Reply by ali-thowfeek** [reply]
I meant, a little help. Because I having a little trouble starting.

And also, with the latest version the trailing \n\n should be fixed. Did you try the latest version.

**Reply by ali-thowfeek** [reply]
Well the trialing spaces for ul and ol still exists. Gotta fix that.

**Reply by nohli** [reply]
Recently, I had to write a few tests for a plutter plugin, so I could start adding a few tests...probably not this or next week, but after the 11th I'll be happy to contribute more!

**Reply by ali-thowfeek** [reply]
> FYI Right now, RichText(text: HTML.toTextSpan(context, string) with a html list (ul) in the string returns \n\n at the end. Probably it would be better without trailing blank space.

I've addressed this on b50cbd5 Hope it fixes it.

> Also, an arrow (& rarr; without the space) isn't converted, although it appears in the package html_unescape

It was small mistake, i have imported `html_unescape_small.dart` instead of the full `html_unescape.dart`
I've fixed it on 45d8913

**Reply by ali-thowfeek** [reply]
@nohli I've published v2.02  (145d74b) with the above issues addressed. Also, made to remove all trailing space. Not just the last three as you can see here 1c1f089. 
Would you be kind enough to edit these changes to null safety for the pre-release when you're free.

**Reply by nohli** [reply]
Actually, all changes were null safe already (if you're using flutter beta, the analyzer will automatically show if something doesn't fit - which you can easily correct, like adding a default value or required, or making the variable nullable with adding a question mark after the type).
I had a hard time merging, so I quickly committed the changes.

Do you have an idea how to work on the indentation of ol and ul tags?
(maybe just add some indentation and bullet point before the element?)

(FYI wanted to push to my fork and then create a pull request, but accidentally pushed to the pre-release-nnbd branch, should be fine)

**Reply by ali-thowfeek** [reply]
Well I prefer you pushing than making a pull request. Thank you for that 

And as for the bullet points and indentation, I was trying using widget span last year. But I couldn't get it done in a neat fashion as far as I can remember. You can see that that I have commented my code approach in `internals.dart` I'll have to look into it again. It might be possible with widget span.

**Reply by nohli** [reply]
@ali-thowfeek Also added lint and refactored the code. You could publish the new prerelease to pub.dev.

**Reply by ali-thowfeek** [reply]
Sorry for the delay. I had to do something about switching flutter channels every now and then. I've got it sorted out. But it took a while. Published  v3.0.0-nullsafety.4 Thank you for the support.

**Comment by ali-thowfeek** [comment]
@kmcgill88 thank you for offering to lend a hand with the tests. We thought of moving ahead with the null safe version of the package. So if you are starting, We'd be glad if you could clone the [nnbd branch](https://github.com/ali-thowfeek/simple_html_css_flutter/tree/pre-release-nnbd) and work on that.
